dev-tools/xtask/src/a4x2_package.rs

```diff
@@ -0,0 +1,382 @@
+    }
+
+    // This needs to happen last because it messes with the working tree in a
+    // way that end-to-end tests doesnt like when building
+    build_a4x2(&sh, &env)?;
+
+    create_output_artifact(&sh, &env)?;
+
+    Ok(())
+}
+
+/// Clone local working tree into work dir. This is nabbed from
+/// buildomat-at-home
+fn prepare_source(sh: &Shell, env: &Environment) -> Result<()> {
+    cmd!(sh, "banner 'prepare'").run()?;
+
+    // Get a ref we can checkout from the local working tree
+    let treeish = {
+        let _popdir = sh.push_dir(&env.src_dir);
+        let mut treeish = cmd!(sh, "git stash create").read()?;
+        if treeish.is_empty() {
+            // nothing to stash
+            eprintln!("Nothing to stash, using most recent commit for clone");
+            treeish = cmd!(sh, "git rev-parse HEAD").read()?;
+        }
+        assert_eq!(
+            treeish.len(),
+            40,
+            "treeish should be a 40-character commit hash"
+        );
+        assert!(
+            treeish.chars().all(|c| c.is_ascii_hexdigit()),
+            "treeish should be a 40-character commit hash"
+        );
+        treeish
+    };
+
+    // Clone the local source tree into the packaging work dir
+    let _popdir = sh.push_dir(&env.omicron_dir);
+    cmd!(sh, "git init").run()?;
+
+    let src_dir = &env.src_dir;
+    cmd!(sh, "git remote add origin {src_dir}").run()?;
+
+    cmd!(sh, "git fetch origin {treeish}").run()?;
+    cmd!(sh, "git checkout {treeish}").run()?;
+
+    // XXX is this needed in CI?
+    cmd!(sh, "./tools/install_builder_prerequisites.sh -yp").run()?;
+
+    Ok(())
+}
+
+fn build_end_to_end_tests(sh: &Shell, env: &Environment) -> Result<()> {
+    // I'm not confident this does what it should, because I don't know much
+    // about the end to end tests. This is just going from what the shell script
+    // did.
+    cmd!(sh, "banner 'end to end'").run()?;
+    let _popdir = sh.push_dir(&env.omicron_dir);
+
+    let cargo = &env.cargo;
+    cmd!(sh, "{cargo} build -p end-to-end-tests --bin commtest --bin dhcp-server --release").run()?;
+
+    let end_to_end_dir = env.out_dir.join("end-to-end-tests");
+    sh.create_dir(&end_to_end_dir)?;
+    sh.copy_file("target/release/commtest", &end_to_end_dir)?;
+    sh.copy_file("target/release/dhcp-server", &end_to_end_dir)?;
+
+    Ok(())
+}
+
+/// Generate a bundle with the live tests included and ready to go.
+fn build_live_tests(sh: &Shell, env: &Environment) -> Result<()> {
+    cmd!(sh, "banner 'live tests'").run()?;
+    let _popdir = sh.push_dir(&env.omicron_dir);
+
+    // We need nextest available, both to generate the test bundle and to
+    // include in the output tarball to execute the bundle on a4x2.
+
+    // XXX right now we only download a new nextest in CI, and we use
+    // the user's locally installed nextest otherwise. I'm inclined to
+    // say we should always download a fresh one for consistency.
+    // thoughts?
```

Comment:
```suggestion
    // XXX right now we only download a new nextest in CI, and we use
    // the user's locally installed nextest otherwise. I'm inclined to
    // say we should always download a fresh one for consistency.
    // thoughts?
```

I'd be inclined to have this tool assume that the user has already set up nextest however they want.  That's what the rest of Omicron currently does, I believe.  If CI needs to download it, that's fine, but it should do it as a separate step before invoking this tool (which I think is what it does for the existing CI jobs).

Reply:
Agreed -- CI already downloads nextest. I'd just package up whatever nextest is available in the `PATH`.